### PR TITLE
Use real proposer for subnet rental proposal

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

When we added `propose-to-rent-subnet` to `test-proposal-payload`, `ic-admin` had a bug where `propose-to-rent-subnet` only worked with `--test-neuron-proposer`. The bug is now fixed.

# Changes

Remove the work-around and use the same method as for other proposals.

# Tests

The test still passes.

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: scripts/nns-dapp/test-proposal-payload [hunk 1]
```diff
@@ -203,17 +203,6 @@ run_ic_admin() {
     grep -E '^proposal [0-9]+$' | sed 's/proposal //'
 }
 
-# Work-around for when `run_ic_admin` gives this error:
-# Specifying a secret key or HSM is only supported for methods that interact with NNS handlers.
-run_ic_admin_with_test_neuron() {
```
Comment (mstrasinskis) [inline scripts/nns-dapp/test-proposal-payload:208]: How about we keep this function in case we need it in the future?

**Review by mstrasinskis — APPROVED** [review]
Thanks!

File: scripts/nns-dapp/test-proposal-payload (hunk 1 above)
Comment (dskloetd) [reply scripts/nns-dapp/test-proposal-payload:208]: It's very easy to make again and might get out of sync if we keep it around. I'd rather delete it.

**Comment by dskloetd** [comment]
It seems that CI doesn't have the fixed version of `ic-admin` yet. I'll update and merge this after sndemo is updated next week.

**Pull request merged**